consensus/src/liveness/proposal_generator.rs

```diff
@@ -285,10 +302,33 @@ impl ProposalGenerator {
 
             let voting_power_ratio = proposer_election.get_voting_power_participation_ratio(round);
 
-            let (max_block_txns, max_block_bytes, proposal_delay) = self
+            let (mut max_block_txns, mut max_block_bytes, mut proposal_delay) = self
                 .calculate_max_block_sizes(voting_power_ratio, timestamp)
                 .await;
 
+            let sys_txn_pull_timer = Instant::now();
```

Comment:
the timer is not used?

Reply:
it's used like this:
```
// calculate proposal_delay.
// ...
let sys_txn_pull_timer = Instant::now();
// ...
// pull sys txns.
// ...
proposal_delay = proposal_delay.saturating_sub(sys_txn_pull_timer.elapsed());
// ...
if !proposal_delay.is_zero() {
    tokio::time::sleep(proposal_delay).await;
}
```

Reply:
i don't have the full details about `proposal_delay`. Here I'm only trying not change the sleep behavior.